src/components/ArticleDetails.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Card, CardBody, CardImg, CardText, CardTitle } from "reactstrap";
+import { getArticleDetails } from '../api/api';
+import Loading from './Loading';
+
+export default function ArticleDetails() {
+    const { article_id } = useParams();
+    const [article, setArticle] = useState([]);
+    const [isLoading, setIsLoading] = useState(true);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        setError(null);
+        setIsLoading(true);
+        getArticleDetails(article_id)
+            .then((article) => {
+                setArticle(article);
+                setIsLoading(false);
+            })
+            .catch((err) => {
+                setIsLoading(false);
+                setError(true);
```

Comment:
If the API request fails, error is a boolean, it may be more useful to pass information to the Error component.

Reply:
On a side note, the Error component doesn't actually exist yet which would normally cause an error in the code. In this case it doesn't as `Error` is a class and so there is no reference error and React tries to turn it into an element.

Reply:
Fixed